src/sol_binder/nonce/base.py

```diff
@@ -58,13 +62,34 @@ def sync_from_chain(self, new_accounts: List[HexAddress] = tuple()):
     def _sync_from_chain(self, account: HexAddress):
         return self._set(account, self.__w3.eth.get_transaction_count(account))
 
-    def get_and_increment(self, account: HexAddress) -> Nonce:
-        nonce = self._get_and_increment(account)
-        self._get_logger().debug(f"Nonce for {account}: {nonce}")
-        return nonce
+    @contextmanager
+    def _lock(self):
```

Comment:
Optional suggestion
IMO a better api would be implementing this as a private like so:
```
@contextmanager
def __lock(self):
   self._acquire_lock()
   yield...
   self._release_lock():

def _acquire_lock(self):
  raise NotImplementedError()

def _release_lock(self):
  raise NotImplementedError()
```